src/Agent.Sdk/Util/PlatformUtil.cs

```diff
@@ -75,6 +85,36 @@ public static bool RunningOnRHEL6
             }
         }
 
+        public static string GetSystemId()
```

Comment:
This can be simplified a bit
```
        public static string GetSystemId()
        {
            return PlatformUtil.HostOS switch
            {
                PlatformUtil.OS.Linux => GetLinuxId(),
                PlatformUtil.OS.OSX => "MacOS",
                PlatformUtil.OS.Windows => GetWindowsId(),
                _ => null
            };
        }
```